modules/images/dominant-color/class-dominant-color-image-editor-imagick.php

```diff
@@ -63,8 +61,25 @@ public function has_transparency() {
 		}
 
 		try {
-			// Check if the image has an alpha channel if true, set to has_transparent to true.
-			return (bool) $this->image->getImageAlphaChannel();
+			// Check if the image has an alpha channel if false, then it can't have transparency so return early.
+			if ( ! $this->image->getImageAlphaChannel() ) {
+				return false;
+			}
+
+			// Walk through the pixels and look transparent pixels.
+			$w = $this->image->getImageWidth();
+			$h = $this->image->getImageHeight();
+			for ( $x = 0; $x < $w; $x++ ) {
+				for ( $y = 0; $y < $h; $y++ ) {
+					$pixel = $this->image->getImagePixelColor( $x, $y );
+					$color = $pixel->getColor();
+					if ( $color['a'] > 0 ) {
+						return true;
+					}
+				}
+			}
+			return false;
```

Comment:
Why is this change needed?

Reply:
I believe that an image just having an alpha channel doesn't mean it has any transparency so I added a test to check to look actual transparent pixels